pyvista/plotting/colors.py

```diff
@@ -987,3 +1000,66 @@ def get_cmap_safe(cmap):
         cmap = ListedColormap(cmap)
 
     return cmap
+
+
+def get_default_cycler():
+    """Return the default color cycler (matches matplotlib's default)."""
+    try:
+        from cycler import cycler
+    except ImportError:
+        raise ImportError('cycler not installed. Please install matplotlib.')
+    return cycler('color', matplotlib_default_colors)
+
+
+def get_hexcolors_cycler():
+    """Return a color cycler for all of the available hexcolors.
+
+    See ``pyvista.plotting.colors.hexcolors``.
+    """
+    try:
+        from cycler import cycler
+    except ImportError:
+        raise ImportError('cycler not installed. Please install matplotlib.')
+    return cycler('color', hexcolors.keys())
+
+
+def get_matplotlib_theme_cycler():
+    """Return matplotlib's current theme's color cycler."""
+    try:
+        import matplotlib.pyplot as plt
+    except ImportError:
+        raise ImportError('Please install matplotlib.')
+    return plt.rcParams['axes.prop_cycle']
+
+
+def get_cycler(color_cycler):
+    """Return a color cycler based on the input value.
+
+    The value must be either a list of color-like objects,
+    or a cycler of color-like objects. If the value passed is a single
+    string, it must be one of:
+
+    * ``'default'`` - Use the default color cycler (matches matplotlib's default)
+    * ``'matplotlib`` - Dynamically get matplotlib's current theme's color cycler.
+    * ``'all'`` - Cycle through all of the available colors in ``pyvista.plotting.colors.hexcolors``
```

Comment:
I'm wondering if we should enable others. Perhaps we could use the discrete colormaps from matplotlib like `tab:20`, etc, as colorcyclers?

I'm sure there are a ton of different palettes we should enable here.

Perhaps, I make it so any matplotlib colormap can be used?

Reply:
Another possibility is using the various color schemes provided by VTK, accessible through the [vtkColorSeries class](https://vtk.org/doc/nightly/html/classvtkColorSeries.html). This is already used by the Charts API:
https://github.com/pyvista/pyvista/blob/ce90705534df7739532d893cebcd1d9673a23c84/pyvista/plotting/charts.py#L1660-L1895

Reply:
Ah, I didn't know about these! I'll toy with these and see how they might fair. If so, I may wan't to move these definitions to `colors.py` and out of the `charts` module

Reply:
Ok great! I think it would be cool to have one 'color cycler' representation that can be used throughout the whole pyvista library (such that Charts can also leverage the matplotlib cyclers for example), so a conversion to/from a custom `vtkColorSeries` would be useful then! Let me know if I can help :)
Please note that if you move the `COLOR_SCHEMES` dict to the colors module, the following line needs to be updated as well (used to construct an overview table of different color schemes in the documentation):

https://github.com/pyvista/pyvista/blob/6ef1a22458510f2d23a18b13caf179eb1887d9a2/doc/make_tables.py#L243

Reply:
Okay, I added in support for named ColorSeries as color cyclers. I think all of this could use further improvement and I think there are a few different routes this could go. I'd like to wrap up this PR as is though and optimize later with more thought and intention.

Reply:
ok for me!